### ISSUE TITLE
feature: winbar improvements

### ISSUE THREAD

**Issue by seblj** [freddiehaddad/feline.nvim#26]
### Did you check the docs?

- [X] I have read all the Feline docs

### Is your feature request related to a problem? Please describe.

I have some code in my config that I feel could be integrated into feline for others to easily use

### Winbar theme

Currently, the theme in the setup is for both statusline and winbar, but I really like having a separate background color for winbar. I am proposing adding a `theme` to the winbar setup, and then falling back to the global theme if not specified

### Local winbar

I like to disable winbar on some buffers like a terminal buffer, startify buffer, etc... I feel like having an option to only list the buffer filetypes, and having feline disabling it for those buffers would be awesome!

I just wanted to open an issue first to hear if you would be open to these changes. I would be happy to implement these changes myself in that case🙌

### Describe alternatives you've considered

Having it as it is in my config

**Comment by freddiehaddad** [comment]
We'd probably want to also add similar support for the statuscolumn as well, so maybe we can plan around some code reuse through utility/helper functions to avoid code duplication.

**Comment by seblj** [comment]
> How were you thinking to add the theme override?
> 
> Something like: `require('feline').winbar.use_theme()` to be consistent with `require('feline').use_theme()`?

Yes! I thought about adding it as an option in the setup function, and then also something like that for `use_theme`.

> We'd probably want to also add similar support for the statuscolumn as well, so maybe we can plan around some code reuse through utility/helper functions to avoid code duplication.

Yes that's a good idea! I'll look into it!

**Comment by seblj** [comment]
Currently the `use_theme` function still works as before, and sets the colors for the statusline, winbar, and statuscolumn. I am not sure if it did set the colors for statuscolumn before, but I made a change inside `reset_highlights` to make it work for `statuscolumn.use_theme`.

`winbar.use_theme` takes precendence over `use_theme` for winbar, and the same for statuscolumn with `statuscolumn.use_theme`

Also, they all have the `theme` option in their respective setup function

**Comment by freddiehaddad** [comment]
Closing since the changes have been merged. Thanks for the contribution!
